src/renderer/components/dock/dock-tabs.tsx

```diff
@@ -109,9 +110,12 @@ export const DockTabs = ({ tabs, autoFocus, selectedTab, onChangeTab, dockStore
     window.addEventListener("resize", onWindowResize);
 
     // update scroll state if tabs numbers has changed
-    dockStore.onTabsNumberChange(() => {
-      updateStateValues();
-    });
+    reaction(() => dockStore.tabsNumber, updateStateValues);
```

Comment:
Mobx reaction returns `IReactionDisposer` (function), so it must be also included inside returning callback.